flow/project.py

```diff
@@ -1582,14 +1583,18 @@ def _generate_operations(self, cmd, jobs, requires=None):
 
     def _get_pending_operations(self, jobs, operation_names=None):
         "Get all pending operations for the given selection."
-        operation_names = None if operation_names is None else set(operation_names)
+        if operation_names is None:
+            names = patterns = None
+        else:
+            patterns = [name.lstrip('/') for name in operation_names if name.startswith('/')]
+            names = [name for name in operation_names if not name.startswith('/')]
```

Comment:
For consistency with the broad scope of regex engines, should we both start _and end_ with slashes?

Reply:
Is that how it typically works? You can search in `vim` with only a trailing slash.

Reply:
I _think_ it's an intended coincidence that `/` triggers the search command in `vim`, not a way of defining a regex.

On another note, most regular expression engines support any character as a regex "pattern delimiter." [1,2] Is it possible to interpret all operation names as regular expressions, and let "exact string matches with the name of an operation" be a special case of that?

[1] https://en.wikipedia.org/wiki/Regular_expression#Delimiters
[2] http://php.net/manual/en/regexp.reference.delimiters.php

Reply:
Looking at those links, using a delimiter on both ends is probably the best way to go. And slash is commonly used, so not the worst choice.

> Is it possible to interpret all operation names as regular expressions, and let "exact string matches with the name of an operation" be a special case of that?

I considered that, but it would definitely break backwards compatibility. If you had two operations *foo* and *foobar* you would now need to do something like this:
```./project.py run -o foo\$```

Extra akward, because we need to escape the `$`. If we believe that this is case is rare and trumps the use of delimiters, then we should consider it.

Reply:
I think there's a way you can use regexes as exact matches or something like that. Such that `foo` matches only `foo` and `foo.*` matches `foo` and `foobar`. This is definitely possible with one of the functions in `re`, let me know if you want me to figure out a code snippet for this behavior.

Reply:
> I think there's a way you can use regexes as exact matches or something like that. Such that `foo` matches only `foo` and `foo.*` matches `foo` and `foobar`. This is definitely possible with one of the functions in `re`, let me know if you want me to figure out a code snippet for this behavior.

In case you are referring to `re.search()` vs `re.match()` you would be out of luck. `foo` is a match for `foo` and `foobar` in both cases. But it's possible that I'm missing something.

Reply:
https://docs.python.org/3/library/re.html#re.fullmatch
`re.fullmatch`, and it's Python 3.4+ only. 😉

```python
>>> re.fullmatch('foo', 'foo')
<_sre.SRE_Match object; span=(0, 3), match='foo'>
>>> re.fullmatch('foo', 'foobar')
>>> # no match found
```

Reply:
Ok, since we have not dropped Python 2.7 support we would probably need to use this [recipe](https://stackoverflow.com/a/30212799).

I am open to this approach.

Reply:
Done. Works great I think.

Reply:
vim is designed to minimize keystrokes, so yes I'd assume that the keystroke choice was intentional. But in general bracketing the expression is the standard for regexes; I think ed/sed/grep are the closest to canonical sources for that.

Regarding having options for delimiters I don't think that's important. Not all regex flavors support that, so we don't have to. I like the fullmatch though!